packages/web/src/dotlottie.ts

```diff
@@ -310,6 +312,28 @@ export class DotLottie {
       window.requestAnimationFrame(this._animationLoop);
     }
   }
+
+  /**
+   * Adjusts the canvas size based on the device pixel ratio and the size of the canvas element.
+   *
+   */
+  private _adjustCanvasSize(): void {
+    const rect = this._canvas.getBoundingClientRect();
+
+    // If the canvas is too large, we don't want to render at a higher resolution. This is to avoid performance issues.
+    const isLargeCanvas = rect.width > 500 || rect.height > 500;
+
+    const ratio = Math.min(window.devicePixelRatio || 1, isLargeCanvas ? 1 : 2);
+
+    const width = rect.width * ratio;
```

Comment:
getBoundingClientRect() returns the size in pixel unit. so this multiplying looks wrong?

Reply:
Probably, you might need to use getClientRects() for the thorvg canvas and adjust the position by adding the padding/margin using the above getBoundingClientRect().